### ISSUE TITLE
Write tests

### ISSUE THREAD

**Issue by raharth** [ob-algdatii-ss18/leistungsnachweis-firststeptoworlddomination#3]
Write tests for literally everything. 

- updateValue: (Solved)
There is some strange behavior in the numerical values in the value function. They can reach values little above 2 while the max should be 1.

- environment:
check if it properly moves the agent around there was a problem where the agent ran in a different direction then he was supposed to, while entering states he shouldn't be able to reach

- policy:
check if it choses the proper actions, test the basic vanilla policy and the softmax

**Comment by raharth** [comment]
Solved the bug with the leaping ValueTable.

In the first place I implemented the QValue update instead of the ValueFunction update, which lead to the leaping effect.

Since we need a numerical value in the field where the agent actually gets a reward, it still has higher values than 1 in adjunct tiles. Therefore we need to be careful with the learning rate to avoid an agent jumping between the two closest tiles instead of going straight towards the reward.
